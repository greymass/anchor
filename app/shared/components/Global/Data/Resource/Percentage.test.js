import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import GlobalDataResourcePercentage from './Percentage';

const mockProps = {
  color: 'green',
  displayResourcesAvailableSetting: false,
  percentageUsed: 90,
  size: 'large',
  style: { margin: 10 },
  t: () => null,
};

describe('GlobalDataResourcePercentage', () => {
  it('renders correctly', () => {
    const wrapper =
      shallow(<GlobalDataResourcePercentage {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('displays resources available when setting enabled', () => {
    const wrapper = shallow(<GlobalDataResourcePercentage
      {...{ ...mockProps, displayResourcesAvailableSetting: true }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Progress').props().percent).toBe(10);
  });
});
