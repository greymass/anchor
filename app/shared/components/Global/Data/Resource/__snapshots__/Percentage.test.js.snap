// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalDataResourcePercentage displays resources available when setting enabled 1`] = `
<Progress
  color="green"
  label={
    <div
      className="label"
    >
      10.000
      %
       
      <Responsive
        as="span"
        getWidth={[Function]}
        minWidth={800}
      />
    </div>
  }
  percent={10}
  size="large"
  style={
    Object {
      "margin": 10,
    }
  }
/>
`;

exports[`GlobalDataResourcePercentage renders correctly 1`] = `
<Progress
  color="green"
  label={
    <div
      className="label"
    >
      90.000
      %
       
      <Responsive
        as="span"
        getWidth={[Function]}
        minWidth={800}
      />
    </div>
  }
  percent={90}
  size="large"
  style={
    Object {
      "margin": 10,
    }
  }
/>
`;
